### PR TITLE
APR-157: add hosted SFT CLI config

### PR DESCRIPTION
**Pull request by tim0120**
## Summary

Adds the hosted SFT distillation config surface to the Prime CLI.

- parses `loss = "sft"` from hosted training TOML
- parses `[teacher]` and `[teacher.sampling]`
- validates that SFT requires a teacher and RL forbids one
- keeps OPD gated until hosted teacher logprob scoring exists
- serializes `loss` and `teacher` into the `/rft/runs` payload
- narrows teacher sampling to the platform-supported teacher config shape
- updates the generated config template and focused config/API tests

This uses the newer GA-facing `loss`/`teacher` shape and is intended to supersede the older `loss_type`/`teacher_rollout_model` shape in PR #514.

## Validation

- `uv run --with pytest python -m pytest packages/prime/tests/test_rl_config.py packages/prime/tests/test_rl_api.py`
- `uv run ruff check packages/prime/src/prime_cli/api/rl.py packages/prime/src/prime_cli/commands/rl.py packages/prime/tests/test_rl_config.py packages/prime/tests/test_rl_api.py`
- `git diff --check`

Linear: APR-157

---

> [!NOTE]
> **Medium Risk**
> Changes how training runs are configured and submitted to the API (new loss modes and teacher payloads), though OPD is rejected client-side and SFT/RL rules are validated before launch.
> 
> **Overview**
> Adds **hosted SFT distillation** to the Prime CLI and `/rft/runs` client using the GA `loss` / `teacher` shape (replacing older `loss_type` / `teacher_rollout_model` style).
> 
> TOML now supports top-level `loss` (`rl` | `sft`; **OPD blocked** until hosted teacher logprob scoring exists), optional `[teacher]` and `[teacher.sampling]` (narrower than student sampling—no temp scheduler), with validation: SFT requires a teacher, RL must not set one. The generated template documents an SFT example; launch confirmation shows loss and teacher model.
> 
> `RLClient.create_run` sends `loss` only when not default `rl` and includes `teacher` when set; `RLRun` reads `teacher` or `teacherConfig`. Tests cover payload serialization and config edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f1e19f4fb2d70f6d8faa55a09e83974982f60b48. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by tim0120** [comment]
@codex review

File: packages/prime/src/prime_cli/commands/rl.py [hunk 1]
```diff
@@ -385,6 +395,38 @@ def _reasoning_controls_mutually_exclusive(self) -> "SamplingConfig":
...
 
 
+class TeacherSamplingConfig(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    max_tokens: int | None = None
+    temperature: float | None = None
+    repetition_penalty: float | None = None
+    min_tokens: int | None = None
+    seed: int | None = None
```
Comment (eexwhyzee) [inline packages/prime/src/prime_cli/commands/rl.py]: these can get punted since they got [deprecated](https://github.com/PrimeIntellect-ai/prime-rl/commit/083127fe329b2d7aa5f884cdf8950ac7b4b2a76d) from prime-rl

**Review by willccbb — DISMISSED** [review]

**Comment by willccbb** [comment]
@cursoragent decide if any of the codex comments are worth resolving and can be done so easily. fix if so, ignore if not

**Review by eexwhyzee — DISMISSED** [review]

File: packages/prime/src/prime_cli/commands/rl.py (hunk 1 above)
Comment (willccbb) [reply packages/prime/src/prime_cli/commands/rl.py]: @cursoragent please remove

**Review by willccbb — APPROVED** [review]
